cmake/recipes/blas.cmake

```diff
@@ -19,8 +19,11 @@ if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "arm64" OR "${CMAKE_OSX_ARCHITECTURES}" M
     # Use Accelerate on macOS M1
     set(BLA_VENDOR Apple)
     find_package(BLAS REQUIRED)
-else()
-    # Use MKL on other platforms
+elseif(POLYSOLVE_WITH_MKL)
```

Comment:
Why would you ever want the system version over the MKL BLAS if available? MKL BLAS should be faster...

Reply:
When `POLYSOLVE_WITH_MKL=OFF`, I would expect MKL to not be used entirely.